### ISSUE TITLE
[Feature Request] Sorting / filtering of site list

### ISSUE THREAD

**Issue by o-l-a-v** [darkreader/darkreader#682]
It's practically impossible to maintain the list of inverted/ not inverted sites when the numbers grow close to or above 40 sites. I can't find any options regarding sorting or filtering these sites, which makes it really hard and cumbersome to maintain.

If what I'm saying is true, and such functionality is not already in place, I'd wish for better overveiw / filter / sorting for making maintenance of sites easier and faster. (Sort by domain, not **https//** and/or **www**.)

(Maybe some functionality to find and remove duplicates, like "https://google.com" and "google.com", would be usefull too)

**Comment by risualiser** [comment]
I think it'd be more comfortable to use "Not invert listed" mode, if you use Dark Reader on most sites.

**Comment by KerimG** [comment]
This is definitely something that I've been missing in the past multiple times. @rolagg 's suggestion isn't really a solution, because eventually the "Not invert listed" list is also going to grow and will be a PITA to maintain.

I forked the repository and will try to work implement this feature.

**Comment by o-l-a-v** [comment]
> I think it'd be more comfortable to use "Not invert listed" mode, if you use Dark Reader on most sites.

I changed to ivert list only the day I liked your comment, but now that list too have become too long to maintain/ change.

- How does Dark Reader **not** have alphabetical sorting by default? It stuns my mind.
- Also, why can't we have a filter/ search in the list, to only show sites that matches what the user is looking for?

**Comment by RyanWils** [comment]
Hi is anyone working on this issue? I would like to work on it.

**Comment by DianaNites** [comment]
Any updates on this? The site list and extension are practically unusable for me, either list grows far too large to manually scroll through, it's pretty broken and unusable. If it's unexpectedly applying to a site I have no choice but to turn it off entirely.

**Comment by Gusted** [comment]
Hi @DianaNites!

Did you already enable the Preview Design under `Devtools -> Enable Preview Design`? It has a better site-list implementation.

As for the sorting idea. It's a great feature be implemented, but not planned in the nearby feature to be implemented.

Regards,
Gusted

**Comment by DianaNites** [comment]
Yes i'm using the preview design, no it's not any better. Unless i'm missing some new feature, it's still just a flat unsearchable unusable list?

**Comment by o-l-a-v** [comment]
It's crazy how sorting or filtering the list of sites is not even planned to be implemented. Should be the easiest thing. But Firefox Multi-Container suffers from the same, maybe it's a common problem amongst browser extensions.

**Comment by Gusted** [comment]
> It's crazy how sorting or filtering/ searching the list of sites is not even on the roadmap

Because it's not a high requested feature, therefor logically a lower priority to get implemented/fixed.

> Should be the easiest thing to implement. 

Good luck implementing it yourself then, if it so easy. :)

> it's a common problem among browser extensions.

It's rather we needed a site list and that's it with just basic functionality, whereby sorting/filtering didn't exist in the basic functionality.

**Comment by o-l-a-v** [comment]
> > It's crazy how sorting or filtering/ searching the list of sites is not even on the roadmap
> 
> Because it's not a high requested feature, therefor logically a lower priority to get implemented/fixed.
> 
> > Should be the easiest thing to implement.
> 
> Good luck implementing it yourself then, if it so easy. :)
> 
> > it's a common problem among browser extensions.
> 
> It's rather we needed a site list and that's it with just basic functionality, whereby sorting/filtering didn't exist in the basic functionality.

I only know PowerShell, but there sorting stuff is as easy as 123. :)

Do you have a voting system for feature request, like UserVoice for instance? Would be good for tracking popularity, and thus prioritizing.

**Comment by Gusted** [comment]
> Do you have a voting system for feature request, like UserVoice for instance? Would be good for tracking popularity, and thus prioritizing.

For now, the emoji's on the first comment should do it. And [thereby using the native sort function](https://github.com/darkreader/darkreader/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) it's easy to track which open issue has the most 👍🏽. 

> I only know PowerShell, but there sorting stuff is as easy as 123. :)

```typescript
// Ascending Order
siteList.sort((a, b) => return a - b);
// Descending Order
siteList.sort((a, b) => return b - a);
```
Main issue with the UI has to be changed for this and some logic behind the scene so it will stay ordered. I will hack some things together and put a PR for it and see how it ends 🤷🏽

**Comment by o-l-a-v** [comment]
I'll admit I'm slow, but just wanted to share that I just found out that with the preview design enabled, going into the site list you can CTRL + F to find a specific domain/fqdn/pattern. Neat.

![image](https://github.com/darkreader/darkreader/assets/6450056/112c74b5-f7fc-4f3e-a233-cf4193fc5121)

I still dream of the day Dark Reader gets the ability to sort the site list + ability to filter list by search.
